tests(generator): enhance init generator tests

anshumanv opened the pull request:
enhance init generator tests

**What kind of change does this PR introduce?**
tests

**Did you add tests for your changes?**
Yay

**If relevant, did you update the documentation?**
NA

**Summary**
* Added file and content checks in the generated files in init generator, missed that in my old PR.

**Does this PR introduce a breaking change?**
No

**Other information**

jamesgeorge007 commented on packages/generators/__tests__/init-generator.test.ts:
```diff
@@ -12,6 +12,16 @@ describe('init generator', () => {
             useExtractPlugin: 'main',
         });
+
+        // Check that all the project files are generator with the correct name
```
```suggestion
        // Check that all the project files are generated with the correct name
```

anshumanv replied:
Good catch, thanks!

ematipico commented on packages/generators/__tests__/init-generator.test.ts:
```diff
@@ -1,7 +1,7 @@
 import { join } from 'path';
 import { run } from 'yeoman-test';
+const assert = require('yeoman-assert');
```
Use the `import`. You are under TypeScript here

anshumanv replied:
Tried it but it wasn't working with this package. :(

ematipico replied:
That's probably because you didn't install the types

ematipico commented on packages/generators/__tests__/init-generator.test.ts:
```diff
@@ -1,7 +1,7 @@
 import { join } from 'path';
```
You should install also `@types/yeoman-assert`

anshumanv replied:
Right, will make the changes, good catch. 👍

ematipico commented on packages/generators/__tests__/init-generator.test.ts:
```diff
@@ -12,6 +12,16 @@ describe('init generator', () => {
+
+        // Check that all the project files are generated with the correct name
+        const filePaths = ['package.json', 'README.md', 'src/index2.js'];
+        assert.file([...filePaths.map(file => join(outputDir, file))]);
```
I had to skip the other tests that I merged because they were failing. You should use the jest assertion in order to tell to jest that is working. Or checking `yeoman-test` documentation to understand how to integrate it with jest

anshumanv replied:
Okay, sounds fair, will take a look. 👍

anshumanv replied:
Is it a problem with yeoman-assert? I can't seem to find a way to integrate them with jest and the docs are really minimal.

anshumanv replied:
Triggered travis on next again and it seems to pass now. 😟🤔

ematipico replied:
Check the logs locally while you test them (maybe run that single test). There are errors and jest can't recognize any assertion so it's like the test is empty

ematipico replied:
At this point do not use `yeoman-assert` and stick with the normal `expect`

anshumanv replied:
> Check the logs locally while you test them (maybe run that single test). There are errors and jest can't recognize any assertion so it's like the test is empty

I just ran a test which was supposed to fail in yeoman-assert and jest caught it, I think the errors from yeoman-assert are caught by jest on exit.

ematipico commented:
This is from the logs

```
PASS packages/generators/__tests__/loader-generator.test.ts (5.522s)
npm ERR! code ENOENT
npm ERR! syscall open
npm ERR! path /tmp/f9ff699a773feb35ce68ffe10cd55974e6943e1c/my-webpack-plugin/package.json
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, open '/tmp/f9ff699a773feb35ce68ffe10cd55974e6943e1c/my-webpack-plugin/package.json'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/runner/.npm/_logs/2020-02-17T09_25_15_716Z-debug.log

ℹ INFO  For more information and a detailed description of each question, have a look at: https://github.com/webpack/webpack-cli/blob/master/INIT.md
ℹ INFO  Alternatively, run "webpack(-cli) --help" for usage info

npm ERR! code ENOENT
npm ERR! syscall open
npm ERR! path /tmp/166db8056f244410d75c82f0f8f7e83bcb726108/my-loader/package.json
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, open '/tmp/166db8056f244410d75c82f0f8f7e83bcb726108/my-loader/package.json'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/runner/.npm/_logs/2020-02-17T09_25_16_276Z-debug.log
```

anshumanv commented:
> This is from the logs

Thanks! I wonder how it's failing now when it was fine before. Unusual. Anyways, will take a look. 👍

anshumanv commented:
Why is snyk failing? Does anybody have any ideas around it? And what packages is it failing for?

ematipico commented:
I merged the PR where we moved the main cli under lerna. Also, we switched to yarn. You should use that now :)

anshumanv commented:
> I merged the PR where we moved the main cli under lerna. Also, we switched to yarn. You should use that now :)

Thanks for the heads up! Will rebase and do the changes in a while. Thanks!

rishabh3112 commented:
Thanks!

anshumanv commented:
Thanks for the rebase! 😄
